### PR TITLE
34 js read directory [Ready for Review]

### LINKED ISSUE #34: javaScript_read Directory of Files
**Description:**
Implement functionality to process an entire directory of TBE files in JavaScript, loading and parsing each file sequentially. This should include options for aggregating metadata and handling errors to facilitate batch processing.

**Developer Notes:**
- Implement a function to read all files in a specified directory, filtering specifically for TBE files.
- Return parsed data in a structured format (e.g., an array or object) for easy access and downstream use.
- Include options to generate summary reports by consolidating key metadata fields across all processed files.
- Provide clear progress updates and error logging to inform the user of file-processing status, including warnings for non-TBE files or issues encountered during parsing.

**Acceptance Criteria:**
- [x] Functionality to read all TBE files in a directory
- [x]  metadata aggregation
- [x]  skip each non-TBE files and log a warning
- [x]  provide logging or progress updates
- [x] an option to return summary metadata in a JSON or visual format

### PR DESCRIPTION
**Pull request by git-voo**
Fixes #34

**What was changed?**

  - Implemented a JavaScript function to:
  - Read all files in a specified directory.
  - Filter files to process only those with a .tbe extension.
  - Parse and extract metadata from TBE files.
  - Return the metadata as a JSON object.

**Why was it changed?**

This functionality fulfills the requirement in Issue #34, enabling users to retrieve metadata without processing the entire file content, which improves efficiency for metadata-focused operations.

**How was it changed?**

  - Created a javascript_tbe directory with a structured sub-directory utils for modular utility functions.
  - Added code for file reading, filtering, and metadata parsing within their respective files.
  - Tested the implementation by running the entry point script read_TBE.js using Node.js, which successfully processed sample TBE files and returned expected metadata results.

### CODE REVIEW

**Review by DeviPriya-Kolla — CHANGES_REQUESTED** [review]
change the read_TBE.js file naming convention into your java/src/functions/readDirectory.js , since both functions acceptance criteria are entirely different

**Comment by git-voo** [comment]
@DeviPriya-Kolla update, and conflict resolved

File: javascript/src/utils/parseUtils.js
Comment (DeviPriya-Kolla) [inline javascript/src/utils/parseUtils.js:1]: The parseTBEMetaData function only retrieves the basic metadata like filename and size, but the actual AC can be further implemented to parse the actual content of TBE files.

Comment (git-voo) [reply javascript/src/utils/parseUtils.js:1]: @DeviPriya-Kolla  That seems to be outside the scope of this PR/issue. The acceptance criteria only specify metadata aggregation, in addition to reading and filtering out non-TBE files, which is all implemented now.

Comment (DeviPriya-Kolla) [reply javascript/src/utils/parseUtils.js:1]: I really appreciate and grateful for your implementation outside the scope.

**Review by DeviPriya-Kolla — APPROVED** [review]

**Review by DeviPriya-Kolla — APPROVED** [review]
lgtm:

**Pull request merged**